src/app/CommandSender.cpp

```diff
@@ -128,6 +128,30 @@ CHIP_ERROR CommandSender::SendCommandRequestInternal(const SessionHandle & sessi
     return SendInvokeRequest();
 }
 
+void CommandSender::CreateBackupForRequestRollback(RollbackData & aRollbackData)
+{
+    VerifyOrReturn(mBufferAllocated);
+    VerifyOrReturn(mState == State::Idle || mState == State::AddedCommand);
+    VerifyOrReturn(mInvokeRequestBuilder.GetInvokeRequests().GetError() == CHIP_NO_ERROR);
+    VerifyOrReturn(mInvokeRequestBuilder.GetError() == CHIP_NO_ERROR);
+    mInvokeRequestBuilder.Checkpoint(aRollbackData.backupWriter);
```

Comment:
This sets several members on aRollbackData. How about having a `RollbackData::Snapshot` or similar that does everything (including setting rollbackIsValid) and then this code would be:

```cpp
aRollbackData.Snapshot(mInvokeRequestBuilder, mState);
```